lib/input_data_page.dart

```diff
@@ -0,0 +1,141 @@
+import 'calculated_data_page.dart';
+import 'constants.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'input_data_bloc.dart';
+
+class InputDataPage extends StatefulWidget {
```

Comment:
For what reason do you need InputDataPage as Stateful Widget?
Maybe would you like to create your Bloc in initState?

Reply:
Refactored to Stateless widget